### DISCUSSION TITLE
Quarto in Rstudio says Julia is not found

### DISCUSSION THREAD

**Discussion by ghost** [quarto-dev/quarto-cli#6711] [Q&A] [answered]
### Description

Hello, 

I am trying the example from the doc, working on Ubuntu 22.04
````markdown
---
title: "Plots Demo"
author: "Norah Jones"
date: "5/22/2021"
format:
  html:
    code-fold: true
jupyter: julia-1.9.3
---

## Parametric Plots

Plot function pair (x(u), y(u)). 
See @fig-parametric for an example.

```{julia}
#| label: fig-parametric
#| fig-cap: "Parametric Plots"

using Plots

plot(sin, 
     x->sin(2x), 
     0, 
     2π, 
     leg=false, 
     fill=(0,:lavender))
```
````

The output is 
```

Warning in system2("bash", "-l -c 'which julia'", stdout = TRUE) :
  running command ''bash' -l -c 'which julia'' had status 1
sh: 1: NA: not found
Error in julia_setup() : Julia is not found.
Error in julia_setup() : Julia is not found.
```

But Julia is installed in my system. If I open a terminal and type julia I end up in the REPL. 
How do I initialize julia_setup()? 

Thank you

**Comment by cderv** [comment]
I am surprised by the content of your document compared to the error message you get. 

I believe the error is from [JuliaCall](https://github.com/Non-Contradiction/JuliaCall) R package, but your .qmd file contains `jupyter: julia-1.9.3` in YAML so should use Python and Jupyter to run Julia with IJulia. 

Can you share you quarto version ? 
Can you try `quarto render` in Terminal ? 

Can you restart and be sure that your environment is clean before rendering the file ?

**Comment by ghost** [comment]
I rebooted the computer. 
```
quarto --version
1.3.450

```

and the render

```
quarto render MA_mutation_analysis.qmd 
ERROR: Jupyter kernel 'julia-1.9.3' not found.

Error: Jupyter kernel 'julia-1.9.3' not found.
    at jupyterKernelspecFromMarkdown (file:///opt/quarto/bin/quarto.js:53188:35)
    at async quartoMdToJupyter (file:///opt/quarto/bin/quarto.js:53021:36)
    at async createNotebookforTarget (file:///opt/quarto/bin/quarto.js:55428:16)
    at async Object.target (file:///opt/quarto/bin/quarto.js:55221:18)
    at async fileExecutionEngineAndTarget (file:///opt/quarto/bin/quarto.js:58033:20)
    at async renderFiles (file:///opt/quarto/bin/quarto.js:86522:46)
    at async render (file:///opt/quarto/bin/quarto.js:90681:21)
    at async Command.fn (file:///opt/quarto/bin/quarto.js:90856:32)
    at async Command.execute (file:///opt/quarto/bin/quarto.js:8437:13)
    at async quarto (file:///opt/quarto/bin/quarto.js:127545:5)

```
I really don't get why it doesn't find julia

**Reply by mcanouil** [reply]
It's not Julia that is not found, but the Jupyter kernel for Julia, see <https://quarto.org/docs/computations/julia.html#installation>.

Note that, from experience with Conda/Julia, the solution was often to completely trash `~/.julia` and start over.

**Reply by cderv** [reply]
Yes error is now clearer and correctly tied to Jupyter. Follow installation as mentioned above and check that you are using a correct engine that you have setup.

**Reply by ghost** [reply]
Thing is, I scrupulously followed the instructions above. I am gonna start all over again. If it still doesn't work I will use Pluto for Julia. Not related but, I did my studies with Rstudio 10 years ago and it's fantastic all the development that have been made. 

Cheers

EDIT: doesn't work, I also tried the route of an independent Jupyter install... I will use Pluto as said since I don't have too much time to spend finding a solution to this. That being said if this is of interest to you, you can still ask details about my install, I am ok with providing details. 

Thanks.

**Reply by ghost** [reply]
Since I am more stubborn .... 

```
quarto check jupyter

[✓] Checking Python 3 installation....OK
      Version: 3.10.12
      Path: /usr/bin/python3
      Jupyter: 5.3.1
      Kernels: julia-1.9, python3

[✓] Checking Jupyter engine render....OK


```

How can quarto finds jupyter in command line in a bash shell, but not in Rstudio?

**Reply by cderv** [reply, marked as answer]
> Kernels: julia-1.9, python3

This shows you that kernel name is `julia-1.9` and not `julia-1.9.3` as you put in your YAML header in .qmd document. You should try with correct name in 
```yaml
jupyter: julia-1.9
```

**Comment by ghost** [comment]
> > Kernels: julia-1.9, python3
> 
> This shows you that kernel name is `julia-1.9` and not `julia-1.9.3` as you put in your YAML header in .qmd document

Indeed, just figured it out as you typed. Strange because it's Julia 9.3 that is installed. thanks for the support.
